irods/connection.py

```diff
@@ -325,7 +325,15 @@ def disconnect(self):
                     self.socket = self.socket.unwrap()
                 except AttributeError:
                     pass
-                self.socket.shutdown(socket.SHUT_RDWR)
+                try:
+                    # On MacOS this can result in the following error:
+                    # OSError: [Errno 57] Socket is not connected
+                    self.socket.shutdown(socket.SHUT_RDWR)
+                except OSError as e:
+                    import errno
```

Comment:
do we have a strong opinion about where this import line lives?   here or at the top?

Reply:
Keeping it where it is helps to show how narrow its usage is and I think that's a good thing.

@d-w-moore @alanking Any thoughts?

Reply:
- pro for 'up top' - single truth, convention, 'performance'
- pro for 'local / in the function' - easier to 'see', and easier to refactor

don't have to be different here, happy to move it up top.

Reply:
moved it

Reply:
Cool cool.